WordPress/gutenberg
# fix/54877: Allow insertion of Buttons in the Navigation Block

## What?


In this PR, the `directInsert` prop is set to `false` to enable insertion of other supported blocks.

## Why?


It is counter-intuitive to insert a custom link, and then transform it into a Button, instead of adding a Button just how one would using the default inserter. This fix reduces confusion and improves user experience.

## How?


## Testing Instructions





In `trunk`

1. Add a Navigation Block
2. Add custom link to the Nav block.
3. Click the inserter within the Nav Block and see you don't get the option to add the Buttons block.
4. Switch to the fix branch and repeat 1, 2 and 3
5. Observe you get the default inserter to insert the Buttons Block.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/17757960/83ccdaa8-ce53-4142-bccd-eff6417db23f
